### PR TITLE
[SCL-58] Auth0 Auth Backend

### PR DESCRIPTION
**Pull request by gridcell**
Trello Card: https://trello.com/c/tCyNPGwD/58-top-authentication-system

### CODE REVIEW

**Comment by gridcell** [comment]
[TOP authentication system](https://trello.com/c/tCyNPGwD/58-top-authentication-system)

File: src/api/models/base.py [hunk 1]
```diff
@@ -40,21 +52,41 @@ def full_name(self):
...
+
+class Application(BaseModel):
+    name = models.CharField(max_length=100)
+    profile = models.ForeignKey(
+        "Profile", related_name="registered_apps", on_delete=models.CASCADE
+    )
+    client_id = models.CharField(max_length=100, unique=True)
+
+    class Meta:
+        db_table = "applications"
```
Comment (saanobhaai) [inline src/api/models/base.py]: I've been deliberately letting django manage everything -- haven't even looked at pgadmin yet. I'd remove this.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  docker:
+    - image: circleci/python:3.7.3
+  environment:
+    AWS_DEFAULT_REGION: us-east-1
+
+install_dependencies: &install_dependencies
+  run:
+    name: Load Dependencies
+    command: |
+      sudo pip install awscli==1.16.96 boto3==1.9.86 awsebcli==3.14.10 fabric==2.5.0
```
Comment (saanobhaai) [inline .circleci/config.yml]: Still a bunch of aws-specific stuff in here...

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    environment:
+      AWS_DEFAULT_REGION: us-east-1
+    steps:
+      - checkout
+      - setup_remote_docker
+      - *install_dependencies
+      - *build
+      - run:
+          name: Setup
+          command: env > .env && fab buildnocache && fab up  && docker cp ./src/. api_service:/var/projects/webapp && sleep 10 && fab createdatabase && fab migrate
```
Comment (saanobhaai) [inline .circleci/config.yml:29]: glad this worked

**Review by saanobhaai — CHANGES_REQUESTED** [review]
Just a couple of little things... and, might be worth checking that the auth0 stuff reflects latest templates/recipes, and not just copied from mermaid (I'm thinking of the recent Collect-side syncing you did)?

**Pull request merged**